### DISCUSSION TITLE
Checkout a PR branch without resetting anything

### DISCUSSION THREAD

**Discussion by DamienCassou** [magit/forge#511] [Ideas]
I would like to be able to checkout the already existing local branch corresponding to the PR at point in magit-status. Currently, <key>b f</key> (`M-x forge-checkout-pullreq`) also resets the branch and I don't want that.

Said differently, I would like the same as <key>b b</key> (`M-x magit-checkout`) but with a list of PRs (and the PR at point by default) instead of a list of branches.

**Comment by DamienCassou** [comment]
I volunteer to write a PR if you tell me what keybinding you would like.

**Reply by tarsius** [reply]
I never had a need to use a dedicated command to checkout existing pr branches because I've managed to keep the number of prs low and the remaining ones are usually short-lived or regularly rebased. So I haven't thought about this much before.

Now that I have, I think it makes most sense to change this command to just check out an existing branch as-is. If the named branch and the pr-ref are out of sink that is immediately shown in the status buffer and can quickly be changed. Some users might feel differently, e.g., because their workflows don't center around the status buffer, so this should be customizable, using an option with the values t, nil and `ask`.

That's a bit more involved than adding a new command but would you still want to implement it? Or maybe you have another idea how this could be done?

**Reply by DamienCassou** [reply]
Done in https://github.com/magit/forge/pull/519.
